### PR TITLE
KAA-1481: 1-thread callback executor (Java client) causes issue with "nested" callback calls

### PR DESCRIPTION
**Pull request by Serega290696**

### CODE REVIEW

File: client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    Assert.assertEquals("Wrong scheduled executor core pool size",
+        minThreads, actualMinThreads);
+  }
+
+  @Test
+  public void testShutdownExecutors() throws Exception {
+    executorContext = new FlexibleExecutorContext();
+    executorContext.init();
+
+    ExecutorService executorService = executorContext.getLifeCycleExecutor();
```
Comment (Rib47) [inline client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java]: Add other executor services from context.

File: client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    executorContext.stop();
+    executorContext.getTimeunit().sleep(executorContext.getTimeout());
+    Assert.assertTrue(executorService.isShutdown());
```
Comment (Rib47) [inline client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java]: Add assertion for other executor services from context.

**Review by Rib47 — CHANGES_REQUESTED** [review]

**Comment by rasendubi** [comment]
@kaa-leeroy retest this please

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+    maxLifeCycleThreadsIdleMilliseconds = DEFAULT_MAX_THREAD_IDLE_MILLISECONDS;
+    maxCallbackThreadsIdleMilliseconds = DEFAULT_MAX_THREAD_IDLE_MILLISECONDS;
+    maxApiThreadsIdleMilliseconds = DEFAULT_MAX_THREAD_IDLE_MILLISECONDS;
+  }
+
+  /**
+   * Constructor for Builder pattern.
+   */
+  public FlexibleExecutorContext(FlexibleExecutorContextBuilder builder) {
```
Comment (vtkhir) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java]: Please change the access modifier to _private_ in this constructor

**Review by vtkhir — CHANGES_REQUESTED** [review]

**Review by vtkhir — APPROVED** [review]
LGTM

**Pull request merged**